src/core/algorithms/nd/nd_verifier/nd_verifier.cpp

```diff
@@ -0,0 +1,228 @@
+#include "algorithms/nd/nd_verifier/nd_verifier.h"
+
+#include <chrono>
+#include <numeric>
+#include <set>
```

Comment:
Remove unused includes and add necessary explicit includes. For example, I don't see using std::set. Furthermore, I see that you use std::unordered_set, however, it isn't included explicitly.